pydicom/tests/conftest.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2008-2020 pydicom authors. See LICENSE file for details.
+"""Fixtures used in different tests."""
+
+import pytest
+
+from pydicom import config
+
+
+@pytest.fixture
+def enforce_valid_values():
+    value = config.enforce_valid_values
+    config.enforce_valid_values = True
+    yield
+    config.enforce_valid_values = value
```

Comment:
Neat

Reply:
I'm not sure this is working as intended somehow.  I pulled the branch to local and if I edit pydicom.config to set the initial value to True, then a number of tests fail.  On master the tests all passed with the same edit (that actually surprised me a little).

Reply:
Good point - I forgot to check this. And yes, this is actually intended - it worked before because in many test cases the value was just set the False in `setup` or `teardown` , and not reset to the default value, meaning that all following tests will have this value set. If these tests are executed stand-alone, they also fail.
This means that I have to explicitely set the config value where it is needed, which I think is better than the random behavior before (explicit is better than implicit...).

Reply:
Shall be fixed now.

Reply:
I probably said this before - each time I see it, I'm again impressed by how fast the tests now run through!

Reply:
> This means that I have to explicitely set the config value where it is needed, which I think is better than the random behavior before (explicit is better than implicit...).

Agreed.  Thanks for the update - working now.

And yeah, the tests - they used to annoy me every time I had to do some little edit to a PR ... so much better now, and the electrons are much less tired.